src/main/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -407,20 +407,28 @@ public boolean hasIndex() {
      */
     @Override
     public BAMIndex getIndex() {
-        if(!hasIndex())
+        if(!hasIndex()) {
             throw new SAMException("No index is available for this BAM file.");
+        }
         if(mIndex == null) {
-            SamIndexes samIndex = getIndexType();
-            if (samIndex == null) {
-                mIndex = mEnableIndexCaching ? new CachingBAMFileIndex(mIndexStream, getFileHeader().getSequenceDictionary())
-                        : new DiskBasedBAMFileIndex(mIndexStream, getFileHeader().getSequenceDictionary());
-            } else if (samIndex.equals(SamIndexes.BAI)) {
+            final SamIndexes samIndex = getIndexType();
```

Comment:
Can we rename samIndex -> samIndexType so it's more obvious it's an enum?
`SamIndexes` is not a super clear class name.

Reply:
Done. Would like to change the class name too. Maybe some day.